mitodl/micromasters
# Implemented proof of concept for Python Social Auth backend

#### What are the relevant tickets?
fixes #2 

#### What's this PR do?
This PR is a proof of concept to demonstrate that an user can login using EDX as backend.

#### Where should the reviewer start?
`backends/edxorg.py` and `micromasters/settings.py`

#### How should this be manually tested?
1. You need to have a working local instance of EDX. 
2. in EDX create an OAUTH client application for micromasters. NOTE, the Redirect URL does matter: if your micromasters app is at the URL `http://192.168.99.102:8079`, then the Redirect URL MUST BE `http://192.168.99.102:8079/complete/edxorg/`
3. update your `.env` file with the edx instance base url, client id and client secret
4. restart micromaster
5. in micrometers go to the url `/login/edxorg/`
6. verify that a new user is created in the micromasters django admin

#### Any background context you want to provide?
In this PR there are no tests given that it is hard to test the OAuth login at this stage unless we can mock a full OAuth provider.

#### What GIF best describes this PR or how it makes you feel?
![](http://i.giphy.com/Tb2i75AlI926I.gif)
